### ISSUE TITLE
segmentation fault for TRJ dataset 

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#73]
This directly related to this issue in ``pytraj``: https://github.com/Amber-MD/pytraj/issues/807#issuecomment-140903727

``pytraj`` is using ``DataSet_Coords_TRJ`` as engine for ``TrajectoryIterator``. I double-check ``pytraj`` code and don't see anything wrong obviously yet.

I tried with cpptraj and got segmentation fault:

```bash
parm system.prmtop
loadtraj ./trunc.nc name test
crdaction test radgyr out test.dat
```

```bash
cpptraj -i water_issue807.in 

CPPTRAJ: Trajectory Analysis. V16.00b
    ___  ___  ___  ___
     | \/ | \/ | \/ | 
    _|_/\_|_/\_|_/\_|_

| Date/time: 09/16/15  18:26:38
| Available memory: 1686.41 MB

INPUT: Reading Input from file water_issue807.in
  [parm system.prmtop]
	Reading 'system.prmtop' as Amber Topology
  [loadtraj ./trunc.nc name test]
	Reading './trunc.nc' as Amber NetCDF
  [crdaction test radgyr out test.dat]
	Using set 'test'
    RADGYR: Calculating for atoms in mask *.
	* (5808 atoms).
 0% Segmentation fault
```

**Note**: it's OK with CRD dataset and regular ``cpptraj`` workflow

```
parm system.prmtop
trajin trunc.nc
radgyr out test.dat
```

**Comment by hainm** [comment]
this is log file for ``loadcrd``, which is ok

```
$ cat water_issue807.in 
parm system.prmtop
loadcrd ./trunc.nc test
crdaction test radgyr out test.dat
```

```
$ cpptraj -i water_issue807.in 
cpptraj -i water_issue807.in 

CPPTRAJ: Trajectory Analysis. V16.00b
    ___  ___  ___  ___
     | \/ | \/ | \/ | 
    _|_/\_|_/\_|_/\_|_

| Date/time: 09/16/15  18:38:55
| Available memory: 1459.07 MB

INPUT: Reading Input from file water_issue807.in
  [parm system.prmtop]
	Reading 'system.prmtop' as Amber Topology
  [loadcrd ./trunc.nc test]
	Reading './trunc.nc' as Amber NetCDF
Warning: COORDS data sets do not store times.
	Loading trajectory './trunc.nc' as 'test'
----- trunc.nc (1-50, 1) -----
 0% 10% 20% 31% 41% 51% 61% 71% 82% 92% 100% Complete.
  [crdaction test radgyr out test.dat]
	Using set 'test'
    RADGYR: Calculating for atoms in mask *.
	* (5808 atoms).
 0% 10% 20% 31% 41% 51% 61% 71% 82% 92% 100% Complete.
TIME: Total action execution time: 0.0202 seconds.
TIME: Total execution time: 0.1207 seconds.
--------------------------------------------------------------------------------
To cite CPPTRAJ use:
Daniel R. Roe and Thomas E. Cheatham, III, "PTRAJ and CPPTRAJ: Software for
  Processing and Analysis of Molecular Dynamics Trajectory Data". J. Chem.
  Theory Comput., 2013, 9 (7), pp 3084-3095.
```

**Comment by drroe** [comment]
Closed via #76
